Swift Shift/src/mouse/MouseTracker.swift

```diff
@@ -128,57 +141,74 @@ class MouseTracker {
             break
         }
     }
-    
+
     private func moveWindowBasedOnMouseLocation(_ event: NSEvent) {
         let currentMouseLocation = NSEvent.mouseLocation
         let deltaX = currentMouseLocation.x - initialMouseLocation!.x
         let deltaY = currentMouseLocation.y - initialMouseLocation!.y
         let newOrigin = NSPoint(x: initialWindowLocation!.x + deltaX, y: initialWindowLocation!.y - deltaY)
-        
+
         WindowManager.move(window: trackedWindow!, to: newOrigin)
     }
-    
+
     private func convertToWindowCoordinates(_ mouseLocation: NSPoint, windowOrigin: NSPoint) -> NSPoint {
         return NSPoint(x: mouseLocation.x - windowOrigin.x, y: mouseLocation.y - windowOrigin.y)
     }
-    
+
     private func resizeWindowBasedOnMouseLocation(_ event: NSEvent) {
-        
         guard let windowSize = windowSize,
               let initialMouseLocation = initialMouseLocation,
               let initialWindowLocation = initialWindowLocation else {
             return
         }
-        
+
         var newWidth: CGFloat, newHeight: CGFloat, newOrigin: NSPoint
-        
+
         if (shouldUseQuadrants) {
             guard let quadrant = quadrant else { return }
-            
+
             let currentMouseLocation = NSEvent.mouseLocation
             let windowRelativeCurrentMouseLocation = convertToWindowCoordinates(currentMouseLocation, windowOrigin: initialWindowLocation)
-            
+
             let deltaX = windowRelativeCurrentMouseLocation.x - (initialMouseLocation.x - initialWindowLocation.x)
             let deltaY = windowRelativeCurrentMouseLocation.y - (initialMouseLocation.y - initialWindowLocation.y)
-            
+
             newWidth = windowSize.width
             newHeight = windowSize.height
             newOrigin = initialWindowLocation
-            
+
             switch quadrant {
             case .topLeft:
                 newWidth -= deltaX
                 newHeight += deltaY
                 newOrigin.x += deltaX
                 newOrigin.y -= deltaY
+            case .top:
+                newHeight += deltaY
+                newOrigin.y -= deltaY
             case .topRight:
                 newWidth += deltaX
                 newHeight += deltaY
                 newOrigin.y -= deltaY
+            case .left:
+                newWidth -= deltaX
+                newOrigin.x += deltaX
+            case .center:
+                if abs(deltaX) > abs(deltaY) {
+                    newWidth += 2 * deltaX
+                    newOrigin.x -= deltaX
+                } else {
+                    newHeight += 2 * deltaY
+                    newOrigin.y -= deltaY
+                }
```

Comment:
This feels better I think, otherwise it's a bit jarring when switching from vertical to horizontal resizing.
Also this is more in line with how linux.AltSnap does centre-resizing
```suggestion
                newWidth += 2 * deltaX
                newOrigin.x -= deltaX
                newHeight += 2 * deltaY
                newOrigin.y -= deltaY
```

Reply:
Here's an example of AltSnap resizing


https://github.com/pablopunk/SwiftShift/assets/1171825/ad254984-6e2b-44ea-bb60-ebb61bd7954e

Also interesting to note, the grid isn't split into exact thirds, but rather makes the center a bit smaller by default. This is configurable per axis, and defaults to being 24% rather than 33%

Reply:
yeah I've tried it and it makes more sense. Still I don't think I'll ever use this but if u guys feel it's more intuitive, let's go for it.